### DISCUSSION TITLE
Getting very low map during mosaic training in 2 channel training

### DISCUSSION THREAD

**Discussion by ajithvcoder** [ultralytics/ultralytics#2138] [Q&A] [answered]
Hi, 
 I am trying to train depth images (two depth images of single channel are merged and being fed to the training). Each of the depth image contains values from 0 to 2000+ and after training i am able to predict properly also. Lets take a example of 40 epochs by default 30 epochs will be mosaic training and remaining 10 epochs will be normal training, during the first 30 epochs the map is only around 0.00xxx to 0.0xxxx but in the last 10 epochs it is raising from 0.5x to 0.72 map i am not sure why its happening like this. i have attached sample logs for 20 epochs where first 10 is mosaic and second 10 is normal . I tried to adjust normalization values img/=255 to img/=2000 but got similar results . kindly help me with this
version 8.0.24
[snap_logs.txt](https://github.com/ultralytics/ultralytics/files/11283869/snap_logs.txt)

**Comment by ajithvcoder** [comment, marked as answer]
problem was in below line in augment.py file

`img4 = np.full((s * 2, s * 2, img.shape[2]), 114, dtype=np.uint8)`

since it was uint8 values were being clipped to 0 to 255 range when i changed to np.int16 it was working fine in mosaic mode . Any other suggestions to make 2 channel training better kindly let me know

**Reply by glenn-jocher** [reply]
@ajithvcoder hello! It's great to hear that you've identified the issue with the data type clipping your depth image values during mosaic augmentation. Using `np.int16` to accommodate the full range of your depth values is a good solution to prevent clipping.

For further improving two-channel training, consider the following suggestions:

1. **Normalization**: Ensure that your normalization step is correctly set up for the range of your depth values. If your depth values range up to 2000+, you might want to normalize by that value instead of 255.

2. **Data Augmentation**: Experiment with different data augmentation techniques that are suitable for depth images. Some augmentations may not be as effective for depth data as they are for RGB data.

3. **Loss Function**: Evaluate if the loss function you're using is appropriate for your task. Sometimes, customizing the loss function can lead to better results for specific types of data.

4. **Learning Rate**: Adjust the learning rate schedule if necessary. Sometimes a different learning rate progression can help the model learn better from two-channel data.

5. **Model Architecture**: If you continue to see suboptimal results, consider modifying the model architecture to better handle two-channel input.

Remember to monitor your model's performance on a validation set to ensure that these changes lead to actual improvements. For more detailed guidance on training and customizing models, you can refer to our documentation at https://docs.ultralytics.com. Keep up the good work! 🚀
